### PR TITLE
Update runners

### PR DESCRIPTION
**Pull request by IsabelParedes**
> The macos-11 label has been deprecated and will no longer be available after 28/6/2024.

### CODE REVIEW

**Review by anutosh491 — APPROVED** [review]
Looks good

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -57,7 +54,7 @@ jobs:
 
       - name: Build
         shell: bash -l {0}
-        run: make -j ${{ steps.cpu-cores.outputs.count }}
+        run: make -j ${{ runner.os == 'macOS' && 3 || 4 }}
```
Comment (JohanMabille) [inline .github/workflows/main.yml:57]: Why this change?

Comment (JohanMabille) [reply .github/workflows/main.yml:57]: Ah, looks like this GHA is now broken (launching too many compilations in parallel).

Comment (IsabelParedes) [reply .github/workflows/main.yml:57]: It was because the `SimenB/github-actions-cpu-cores@v2` had not been working in a while. `macos-12` is the only one with 3 cores, all the others are 4 now.
